install.sh

```diff
@@ -564,30 +570,255 @@ compare_versions() {
-    # Older Debian/Ubuntu/etc.
-    distro="debian"
-  elif [[ -f "/etc/SuSe-release" ]]; then
-    # Older SuSE/etc.
-    distro="suse"
-  elif [[ -f "/etc/redhat-release" ]]; then
-    # Older Red Hat, CentOS, etc.
-    distro="redhat"
+prepare_environment_for_werf() {
+
+  local override_prepare_environment_for_werf="$1"
+
+  [[ $override_prepare_environment_for_werf == "no" ]] && return 0
+
+  [[ $override_prepare_environment_for_werf == "auto" ]] && prompt_yes_no_skip "Need to install packages for werf?" "yes" || return 0
+
+  if ! is_command_exists git; then
+    install_package git
+  else
+    already_installed git
+  fi
+
+  if ! is_command_exists curl; then
+    install_package curl
+  else
+    already_installed curl
+  fi
+
+  if ! is_command_exists gpg; then
+    install_package gpg
+  else
+    already_installed gpg
+    echo
+  fi
+
+}
+
+prepare_environment_for_buildah() {
+
+  local override_prepare_environment_for_buildah="$1"
+
+  [[ $override_prepare_environment_for_buildah == "no" ]] && return 0
+
+  [[ $override_prepare_environment_for_buildah == "auto" ]] && prompt_yes_no_skip "Need to install packages buildah and set kernel values?" "yes" || return 0
+
+  if ! is_command_exists buildah; then
+    install_package buildah
+  else
+    already_installed buildah
+  fi 
+  # Get the current user's name
+  CURRENT_USER=$(get_user)
+
+  # Create the path /home/<current user>/.local/share/containers if it doesn't exist
+  CONTAINER_PATH="/home/${CURRENT_USER}/.local/share/containers"
+  if [ ! -d "$CONTAINER_PATH" ]; then
+    mkdir -p "$CONTAINER_PATH"
+    log::info "Created directory: $CONTAINER_PATH"
+  fi
+
+  # Ensure the current user has read and write access to the directory
+  chmod u+rw "$CONTAINER_PATH"
+  log::info "Set read and write permissions for $CURRENT_USER in $CONTAINER_PATH"
+  echo
+  
+  if is_command_exists sysctl; then
+  # Check if kernel.unprivileged_userns_clone exists
+    if ! [ -z $(sysctl -ne kernel.unprivileged_userns_clone) ]; then
+      KERNEL_SETTING=$(sysctl -ne kernel.unprivileged_userns_clone)
+      if [ "$KERNEL_SETTING" -eq 0 ]; then
+      echo 'kernel.unprivileged_userns_clone = 1' | run_as_root "tee -a /etc/sysctl.conf"
+      run_as_root "sysctl -p"
+      log::info "Updated kernel.unprivileged_userns_clone value"
+      fi
+    fi
+
+    # Check if kernel.apparmor_restrict_unprivileged_userns exists
+    if ! [ -z $(sysctl -ne kernel.apparmor_restrict_unprivileged_userns) ]; then
+      KERNEL_APPARMOR=$(sysctl -ne kernel.apparmor_restrict_unprivileged_userns)
+      if [ "$KERNEL_APPARMOR" -eq 1 ]; then
+        log::info "Set kernel.apparmor_restrict_unprivileged_userns and kernel.apparmor_restrict_unprivileged_unconfined to 0"
+        { echo "kernel.apparmor_restrict_unprivileged_userns = 0" \ &&
+          echo "kernel.apparmor_restrict_unprivileged_unconfined = 0";} | run_as_root "tee -a /etc/sysctl.d/20-apparmor-donotrestrict.conf" \ &&
```

Comment:
Yes.
There is a recommendation from `/lib/sysctl.d/10-apparmor.conf` for `kernel.apparmor_restrict_unprivileged_userns = 1`
```
# If it is desired to disable this restriction, it is preferable to create an
# additional file named /etc/sysctl.d/20-apparmor.conf which will override this
# current file and sets this value to 0 rather than editing this current file
```
If you set this value to 0 in the /etc/sysctl.conf it would be overwrite to 1 after reboot

Reply:
> If you set this value to 0 in /etc/sysctl.conf, it will be overwritten to 1 after a reboot.

I really don't think this is true (not tested myself though). This goes against what described in sysctl manual:
https://man7.org/linux/man-pages/man5/sysctl.conf.5.html#FILES

From my understanding /etc/sysctl.conf has the highest priority of all. And also any files in `/etc/sysctl.d/` that go alphabetically after `20-apparmor-donotrestrict.conf` will override values set in `20-apparmor-donotrestrict.conf`.